cts/lab/CTStests.py

```diff
@@ -1291,6 +1279,55 @@ def __call__(self, node):
                 # Anonymous clones may get restarted with a different clone number
                 pats.append(self.templates["Pat:RscOpOK"] % ("start", ".*"))
 
+        # Fail resource. (Ideally, we'd fail it twice, to ensure the fail count
+        # is incrementing properly, but it might restart on a different node.
+        # We'd have to temporarily ban it from all other nodes and ensure the
+        # migration-threshold hasn't been reached.)
+        if self.fail_resource(rsc, node, pats) is None:
+            return None # self.failure() already called
+
+        return self.success()
+
+    def choose_resource(self, node, resourcelist):
+        """ Choose a random resource to target """
+
+        self.rid = self.Env.RandomGen.choice(resourcelist)
+        self.rid_alt = self.rid
+        (rc, lines) = self.rsh(node, "crm_resource -c", None)
+        for line in lines:
+            if line.startswith("Resource: "):
+                rsc = AuditResource(self.CM, line)
+                if rsc.id == self.rid:
+                    # Handle anonymous clones that get renamed
+                    self.rid = rsc.clone_id
+                    return rsc
+        return None
+
+    def get_failcount(self, node):
+        """ Check the fail count of targeted resource on given node """
+
+        (rc, lines) = self.rsh(node,
+                               "crm_failcount --quiet --query --resource %s "
+                               "--operation %s --interval %d "
+                               "--node %s" % (self.rid, self.action,
+                               self.interval, node), None)
+        if rc != 0 or len(lines) != 1:
+            self.logger.log("crm_failcount on %s failed (%d): %s" % (node, rc,
+                            " // ".join(map(str.strip, lines))
```

Comment:
Parentheses need fixing. Interestingly it broke CI RPM tests on everything except the OpenSUSE machines. Maybe they don't run the CTS lab scripts.

Reply:
Whoops. CI never runs the lab, but the rpms byte-compile the Python, which is what failed. I guess OpenSUSE doesn't do that, at least not with the syntax we have ...